standards/component-documentation.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Component Documentation
+description: This standard defines how components should be documented in their blueprint description.
+---
+
+# Component Documentation
+
+## Overview
+
+This standard defines how components should be documented in their blueprint description. The goal is to provide a consistent and informative format that works for all components. By following this standard, it will be easier for users to understand how to use a component, because the formatting and terminology is consistent with other components.
+
+## Terminology
+
+- :bearing: **Component**: A logic creation intended to be used inside of other larger logic creations. Examples include ALUs, registers, and memory units.
+- :bearing: **Interface**: A set of logic gates that function as input and/or output gates.
+
+## Blueprint Description
+### Short non-technical description
+The blueprint description should start with a short non-technical description of the component. This should be no longer than a paragraph. This description should explain what the component does, and why it is useful. Avoid immediately diving into technical details, because that's for a later section.
+
+### Interface declaration
+Next, we describe what inputs and outputs the component has. This should be done using the following format:
+`<interface gate color> <amount of gates> <input/output> : <short but descriptive name>`
+(see examples below)
+
+### Detailed description
+After the interface declaration, we should provide a more detailed description of the component. This should include how the component works, and how it should be used. This section should be written in a more technical style, and should be more detailed than the short non-technical description. Here you can use the names of the interfaces you declared earlier.
+
+Examples of stuff to discuss are:
+- How to use the component
+- What conventions you used for LSB marking, signed binary number formats, etc.
+- With what delay your component responds to inputs
+- What the limits of the component are, and what happens if you exceed them
+- Things that might cause undefined or buggy behavior
+- Whether or not it is okay to "merge delete" the input and output interfaces to reduce the delay of the component
+- Links to external resources like data sheets or more thorough explanations of the component if they don't fit nicely in a blueprint description
+
+### Contact information, license, and credits
+Finally, the blueprint description should include contact information for the creator of the component (probably you). Preferably, this should include a Discord username, but something like an email or a statement that you want messages via steam is also fine.
+Here you can also give other people permission to use your component, and if and how you want to be credited for it.
+Of course if you used someone else's creation or idea in your component, you should credit them here, even if you got permission from them.
```

Comment:
- As long as we use steam workshop collections, the steam account of the uploader is always available. Other contact information can be optionally added.
- Maybe we should make it so that every creation that is submitted to the SMLC is free to use and modify and when you submit something to the SMLC it is advised to give credit to the original creators of the respective creations that were used.
  Maybe add a paragraph like "this creation was submitted to the SMLC collection, it is free to use and modify, credits are appreciated." to SMLC submissions.